app/domain/operations/eligible/build_eligibility.rb

```diff
@@ -0,0 +1,126 @@
+      # @option opts [Hash]       :timestamps optional timestamps for data migrations purposes
+      # @return [Dry::Monad] result
+      def call(params)
+        values = yield validate(params)
+        evidence_record = yield find_evidence(values)
+        evidence_options = yield build_evidence_options(values, evidence_record)
+        eligibility_options = yield build_eligibility_options(values, evidence_options)
+
+        Success(eligibility_options)
+      end
+
+      private
+
+      def validate(params)
+        errors = []
+        errors << "subject missing" unless params[:subject]
+        errors << "evidence key missing" unless params[:evidence_key]
+        errors << "evidence value missing" unless params[:evidence_value]
+        errors << "effective date missing or it should be a date" unless params[:effective_date].is_a?(::Date)
+        @current_user = GlobalID::Locator.locate(params[:current_user])
+        errors << "current_user missing or not found for #{params[:current_user]}" unless @current_user.present?
+
+        errors.empty? ? Success(params) : Failure(errors)
+      end
+
+      def find_evidence(values)
+        eligibility_rec = values[:eligibility_record]
+        return Success(nil) unless eligibility_rec
+
+        evidence_record = eligibility_rec.evidences.by_key(values[:evidence_key]).max_by(&:created_at)
+        Success(evidence_record)
+      end
+
+      def build_evidence_options(values, evidence_record = nil)
+        options = {}
+        options[:configuration] = values[:evidence_configuration] if values[:evidence_configuration]
+
+        ::Operations::Eligible::BuildEvidence.new(**options).call(
+          values.merge(evidence_record: evidence_record)
+        )
+      end
+
+      def build_eligibility_options(values, evidence_options)
+        options = build_default_eligibility_options(values)
+
+        if options[:evidences].present?
+          index = options[:evidences].index do |e|
+            e[:_id].to_s == evidence_options[:_id].to_s
+          end
+        end
+
+        if index
+          options[:evidences][index] = evidence_options
+        else
+          options[:evidences] = [evidence_options]
+        end
+
+        options[:state_histories] ||= []
+        new_state_history = build_elgibility_state_history(values, options[:evidences])
+        options[:state_histories] << new_state_history
+        options[:current_state] = new_state_history[:to_state]
+
+        Success(options)
+      end
+
+      def build_default_eligibility_options(values)
+        if values[:eligibility_record]&.persisted?
+          # returns existing record hash
+          return(values[:eligibility_record].serializable_hash.deep_symbolize_keys)
+        end
+
+        {
+          title: configuration.title,
+          key: configuration.key,
+          grants: build_grants
+        }
+      end
+
+      def build_grants
+        configuration.grants.compact.collect do |value_pair|
```

Comment:
Looks like `value_pair` is symbol. Then when we take `value_pair[0]` and `value_pair[1]`
then params is equal `{:grant_key=>"d", :grant_value=>"e"}`

Are you sure is that what we expect?